client/resource_feature.go

```diff
@@ -74,6 +74,14 @@ func ResourceFeature() *schema.Resource {
 				Optional:    true,
 				Description: "An SQL expression to apply to the result of the feature aggregation.",
 			},
+			"entity_restrictions": {
+                Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of entity Id's that the feature is restricted to.",
+				Elem: &schema.Schema{
+                	Type: schema.TypeString,
+                },
```

Comment:
You may need to be more careful with regards to the empty list.

As it stands, if you set an empty list on the resource it looks like it will serialize `[]` in the json, which is invalid as per the scala types.

Unfortunately I don't think MinItems will do the right thing (but test it maybe?), so it might need handling in `buildFeature`.

Reply:
Actually I'm having trouble getting this to work even if it's not specified at all. It's still encoding an empty list.

Reply:
I think handling in buildFeature with a length comparison instead of a nil comparison would be the most robust.
```
restrictions := d.Get("entity_restrictions").([]interface{})
if len(restrictions) > 0 {
  ...
}
```

Reply:
done, changed to length comparison and assigning `nil` entity restrictions if it empty list so that it gets picked up by `omitempty`